src/transformers/modeling_utils.py

```diff
@@ -904,8 +904,8 @@ def _load_state_dict_into_meta_model(
                         param = param[:].to(param_casting_dtype)
                     module.load_state_dict(
                         {param_type: param[:].to(param_device)},
-                        False,
-                        True,
+                        strict=False,
+                        assign=False,
```

Comment:
```suggestion
                        assign=True,
```
we need to assign to get speed boost / leverage meta device

Reply:
oke, then we will need to either cast params to same `dtype` as module or add a more involved logic of passing `dtype` for composite model. Personally, I think the second way is will complicate things even more

Otherwise any time one wants to load different dtypes for each backbones with accelerate (`device_map` or `low_cpu_mem-usage`), one will get only one `dtype` for the whole model